src/main/java/seedu/address/logic/commands/wallet/WalletEditCommand.java

```diff
@@ -0,0 +1,221 @@
+package seedu.address.logic.commands.wallet;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliPrefix.WALLET_COMMAND_TYPE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.transaction.Amount;
+import seedu.address.model.transaction.Date;
+import seedu.address.model.transaction.Description;
+import seedu.address.model.transaction.Expense;
+import seedu.address.model.transaction.Income;
+import seedu.address.model.transaction.Transaction;
+
+/**
+ * Edits the details of an existing income/expense in the address book.
+ */
+public class WalletEditCommand extends Command {
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the transaction identified "
+            + "by the index number used in the displayed transaction list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_NAME + "NAME "
+            + PREFIX_AMOUNT + "AMOUNT "
+            + "[" + PREFIX_DATE + "DATE] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + WALLET_COMMAND_TYPE + " "
+            + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Duck Rice "
+            + PREFIX_AMOUNT + "3.80";
+
+    public static final String MESSAGE_EDIT_TRANSACTION_SUCCESS = "Edited Transaction: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+
+    private final Index index;
+    private final EditTransactionDescriptor editTransactionDescriptor;
+
+    /**
+     * @param index                     of the person in the filtered transaction list to edit
+     * @param editTransactionDescriptor details to edit the transaction with
+     */
+    public WalletEditCommand(Index index, EditTransactionDescriptor editTransactionDescriptor) {
+        requireNonNull(index);
```

Comment:
You can shorten this to `requireAllNonNull(obj1, obj2, ...)`

Reply:
Woah thanks! Didn't know about this.